iceoryx_examples/iceperf/iceoryx.cpp

```diff
@@ -68,24 +69,37 @@ void Iceoryx::shutdown() noexcept
 
 void Iceoryx::sendPerfTopic(uint32_t payloadSizeInBytes, bool runFlag) noexcept
 {
-    auto sendSample = static_cast<PerfTopic*>(m_publisher.allocateChunk(payloadSizeInBytes, true));
-    sendSample->payloadSize = payloadSizeInBytes;
-    sendSample->run = runFlag;
-    sendSample->subPackets = 1;
-
-    m_publisher.sendChunk(sendSample);
+    m_publisher.loan(payloadSizeInBytes).and_then([&](iox::popo::Sample<void>& sample) {
+        auto sendSample = static_cast<PerfTopic*>(sample.get());
```

Comment:
Strictly speaking you have undefined behavior here. You cast something to a pointer without calling the constructor and the contents of the sample are arbitrary.
I think you should do a placement new like this
```cpp
new (sendSample) PerfTopic{payloadSizeInBytes, 1U, runFlag};
```

Reply:
Or use the `TypedPublisher` which does this for you.

Reply:
So, we keep it the way it is or use the placement new with c'tor?